### ISSUE TITLE
Cannot authenticate with pawoo.net

### ISSUE THREAD

**Issue by meyou69** [tuskyapp/Tusky#2534]
When entering the instance name "pawoo.net" and pressing the ”LOGIN WITH MASTODON" button, the application takes me to an empty page afterwards with just the back button in it and then, after like 2 seconds, it turns me back to the "Which instance?" form page. Basically, I cannot login to my pawoo.net account

* * * *
- Tusky Version: 17.0
- Android Version: 11
- Android Device: OnePlus 5T
- Mastodon instance (if applicable): pawoo.net

- [ x ] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by connyduck** [comment]
Can you try with 18.0 beta 1 please? We made some improvements to the login that could fix this, at least it should give you an error message now.

**Comment by meyou69** [comment]
I would love to but can you please explain how to install that particular version? Both the Play store and F-Droid only have version 17 available

**Comment by meyou69** [comment]
Never mind, I eventually found the Tusky Test app on the Play store, which, I believe is at version 18.
I installed it but I'm getting a different error now. Please see the attached screenshot
![Screenshot_20220516-162155_Tusky_Test](https://user-images.githubusercontent.com/92801613/168602584-c6916038-2860-4313-a5da-fe138b055c30.png)

**Comment by lost-geographer** [comment]
Hello, I might have a similar problem : I got a "could not load the login page" error, while I was trying to login to the framapiaf.org server. I tried the 18 and 19b version of tusky. I have Android 11.

**Comment by lost-geographer** [comment]
> Hello, I might have a similar problem : I got a "could not load the login page" error, while I was trying to login to the framapiaf.org server. I tried the 18 and 19b version of tusky. I have Android 11.
> 
> Edit : I tried another server like pouet.chapril.org and it works. It might be a server misconfigurarion ?

I found a solution : actually, I use a private ad-blocking dns that blocks the stats.framasoft.org domain (which hosts an instance of the Matomo analytics software), which prevents the login page from loading. Reverting temporarily to the default DNS while connecting to the framapiaf.org server solved the issue. The stats.framasoft.org is blocked also by ad-block lists like  StevenBlack, Energized Protection and oisd full.

**Comment by connyduck** [comment]
> private ad-blocking dns

Yes that explains it. @charlag do we have a way to not fail when a non essential call (like tracking) fails? It seems `onReceivedError` is called everytime no matter which call fails. If not, maybe better leave the webview open in all cases so cases like this can still log in?

**Comment by charlag** [comment]
@connyduck 

>Note that unlike the deprecated version of the callback, the new version will be called for any resource (iframe, image, etc.), not just for the main page. Thus, it is recommended to perform minimum required work in this callback.

yeah I think we can check `request.url()` and only fail if it's this URL that fails. It might be more tricky with redirects but maybe it's not as important corner case
